repository_service_tuf_worker/signer.py

```diff
@@ -1,22 +1,83 @@
 # SPDX-FileCopyrightText: 2024 Repository Service for TUF Contributors
 #
 # SPDX-License-Identifier: MIT
+import os
+from pathlib import Path
+from typing import Optional
 
+from cryptography.hazmat.primitives.serialization import load_pem_private_key
 from dynaconf import Dynaconf
 from securesystemslib.signer import (
     SIGNER_FOR_URI_SCHEME,
     CryptoSigner,
     Key,
+    SecretsHandler,
     Signer,
 )
 
 from repository_service_tuf_worker.interfaces import IKeyVault
 
+
+class FileNameSigner(CryptoSigner):
+    """File-based signer implementation.
+
+    Provide method to load **unencrypted** PKCS8/PEM private key from file.
+
+    File path is constructed by joining base path in envrionment variable
+    ``RSTUF_ONLINE_KEY_DIR`` with file in ``priv_key_uri``.
+
+    NOTE: Make sure to use the secrets management service of your deployment
+    platform to protect your private key!
+
+    Example::
+
+        RSTUF_ONLINE_KEY_DIR (env) "/run/secrets"
+        priv_key_uri (arg): "fn:foo"
+
+        File path: "/run/secrets/foo"
+
+    Raises:
+        KeyError: RSTUF_ONLINE_KEY_DIR environment variable not set
+        OSError: file cannot be loaded
+        ValueError: uri has no filenmae, or private key cannot be decoded,
+                or type does not match public key
+        cryptography.exceptions.UnsupportedAlgorithm: key type not supported
+
+    """
+
+    SCHEME = "fn"
+    DIR_VAR = "RSTUF_ONLINE_KEY_DIR"
+
+    @classmethod
+    def from_priv_key_uri(
+        cls,
+        priv_key_uri: str,
+        public_key: Key,
+        secrets_handler: Optional[SecretsHandler] = None,
+    ) -> "FileNameSigner":
+        _, _, file_name = priv_key_uri.partition(":")
+        if not file_name:
+            raise ValueError(
+                f"bad uri: expected '{cls.SCHEME}:<file name>', "
+                f"'got {priv_key_uri}'"
+            )
+
+        dir_ = os.environ[cls.DIR_VAR]
```

Comment:
RSTUF stores all `RSTUF_*` environment variables in Dynaconf as settings.
It is used by the `SignerStore`, but cannot be used in the custom signer `FileNameSigner` will use it outside the RSTUF instance.

Is there a way to keep it centralized in the RSTUF dynaconf settings?

Reply:
See code snippet in [comment below](https://github.com/repository-service-tuf/repository-service-tuf-worker/pull/455#issuecomment-1940942231).